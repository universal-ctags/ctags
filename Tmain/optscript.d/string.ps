(--------------- string ---------------) ==
(32 32 32 ->)
3 string { } forall pstack clear
(--------------- _strstr ---------------) ==
((abcdefgab) 0 true->)
(abcdefgab) (ab) _strstr pstack clear

((abcdefgab) 5 true ->)
(abcdefgab) (fg) _strstr pstack clear

((abcdefg) 6 true ->)
(abcdefg) (g) _strstr pstack clear

((abcdefg) 0 true ->)
(abcdefg) () _strstr pstack clear

((abcdefg) false ->)
(abcdefg) (z) _strstr pstack clear

(--------------- _strrstr ---------------) ==
((abcdefgab) 7 true->)
(abcdefgab) (ab) _strrstr pstack clear

((abcdefgab) 5 true ->)
(abcdefgab) (fg) _strrstr pstack clear

((abcdefg) 6 true ->)
(abcdefg) (g) _strrstr pstack clear

((abcdefg) 7 true ->)
(abcdefg) () _strrstr pstack clear

((abcdefg) false ->)
(abcdefg) (z) _strrstr pstack clear

(--------------- _strchr/_strrchr ---------------) ==

/char { 0 get } def

((aba) 0 true ->)
(aba) (a) char _strchr pstack clear

((aba) 2 true ->)
(aba) (a) char _strrchr pstack clear

((aba) 1 true ->)
(aba) (b) char _strchr pstack clear

((aba) 1 true ->)
(aba) (b) char _strrchr pstack clear

((aba) false ->)
(aba) (c) char _strchr pstack clear

((aba) false ->)
(aba) (c) char _strrchr pstack clear

(--------------- _strpbrk ---------------) ==

((a.b;c) 1 true ->)
(a.b;c) (;.) _strpbrk pstack clear

((a.b;c) false ->)
(a.b;c) (,-) _strpbrk pstack clear

(--------------- _strrpbrk ---------------) ==

(x/abc.d|./) =
(x/abc.d) (./) _strrpbrk pstack clear

(x/abc.|./) =
(x/abc.) (./) _strrpbrk pstack clear

(x.abc/d|./) =
(x.abc/d) (./) _strrpbrk pstack clear

(x.abc/|./) =
(x.abc/) (./) _strrpbrk pstack clear

(x.abc.d|./) =
(x.abc.d) (./) _strrpbrk pstack clear

(x.abc.|./) =
(x.abc.) (./) _strrpbrk pstack clear

(abc|./) =
(abc) (./) _strrpbrk pstack clear

(|./) =
() (./) _strrpbrk pstack clear

(abc|) =
(abc) () _strrpbrk pstack clear

(|) =
() () _strrpbrk pstack clear
