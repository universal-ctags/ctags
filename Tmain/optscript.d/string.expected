(--------------- string ---------------)
32
32
32
(32 32 32 ->)
(--------------- _strstr ---------------)
true
0
(abcdefgab)
(\(abcdefgab\) 0 true->)
true
5
(abcdefgab)
(\(abcdefgab\) 5 true ->)
true
6
(abcdefg)
(\(abcdefg\) 6 true ->)
true
0
(abcdefg)
(\(abcdefg\) 0 true ->)
false
(abcdefg)
(\(abcdefg\) false ->)
(--------------- _strrstr ---------------)
true
7
(abcdefgab)
(\(abcdefgab\) 7 true->)
true
5
(abcdefgab)
(\(abcdefgab\) 5 true ->)
true
6
(abcdefg)
(\(abcdefg\) 6 true ->)
true
7
(abcdefg)
(\(abcdefg\) 7 true ->)
false
(abcdefg)
(\(abcdefg\) false ->)
(--------------- _strchr/_strrchr ---------------)
true
0
(aba)
(\(aba\) 0 true ->)
true
2
(aba)
(\(aba\) 2 true ->)
true
1
(aba)
(\(aba\) 1 true ->)
true
1
(aba)
(\(aba\) 1 true ->)
false
(aba)
(\(aba\) false ->)
false
(aba)
(\(aba\) false ->)
(--------------- _strpbrk ---------------)
true
1
(a.b;c)
(\(a.b;c\) 1 true ->)
false
(a.b;c)
(\(a.b;c\) false ->)
(--------------- _strrpbrk ---------------)
x/abc.d|./
true
5
(x/abc.d)
x/abc.|./
true
5
(x/abc.)
x.abc/d|./
true
5
(x.abc/d)
x.abc/|./
true
5
(x.abc/)
x.abc.d|./
true
5
(x.abc.d)
x.abc.|./
true
5
(x.abc.)
abc|./
false
(abc)
|./
false
()
abc|
false
(abc)
|
false
()
